src/PrecipitationDecorator.java

```diff
@@ -0,0 +1,10 @@
+public class PrecipitationDecorator extends ClimateDecorator {
+    public PrecipitationDecorator(ClimateObserver decoratedMonitor) {
```

Comment:
i do not see Decorator pattern and i am still do not understand of creating this class